src/eko/output/struct.py

```diff
@@ -0,0 +1,502 @@
+
+        __import__("pdb").set_trace()
+        return content
+
+    @property
+    def theory(self) -> dict:
+        return yaml.safe_load(self.extract(self.path, "theory.yaml"))
+
+    @property
+    def theory_card(self) -> dict:
+        return self.theory
+
+    @property
+    def operator_card(self) -> dict:
+        return yaml.safe_load(self.extract(self.path, "operator.yaml"))
+
+    @classmethod
+    def detached(cls, theory: dict, operator: dict, path: pathlib.Path):
+        """Build the in-memory object alone.
+
+        Note
+        ----
+        This constructor is meant for internal use, backing the usual ones (like
+        :meth:`new` or :meth:`load`), but it should not be used directly, since
+        it has no guarantee that the underlying path is valid, breaking the
+        object semantic.
+
+        Parameters
+        ----------
+        theory : dict
+            the theory card
+        operator : dict
+            the operator card
+        path : str or os.PathLike
+            the underlying path (it has to be a valid object, but it is not
+            guaranteed, see the note)
+
+        Returns
+        -------
+        EKO
+            the generated structure
+
+        """
+        xgrid = interpolation.XGrid(operator["xgrid"])
+        pids = np.array(operator.get("pids", np.array(br.flavor_basis_pids)))
+
+        return cls(
+            path=path,
+            xgrid=xgrid,
+            pids=pids,
+            Q02=operator["Q0"] ** 2,
+            _operators={q2: None for q2 in operator["Q2grid"]},
+            configs=Configs.from_dict(operator["configs"]),
+            rotations=Rotations.default(xgrid, pids),
+            debug=Debug.from_dict(operator.get("debug", {})),
+        )
+
+    @classmethod
+    def new(cls, theory: dict, operator: dict, path: Optional[PathLike] = None):
+        """Make structure from runcard-like dictionary.
+
+        This constructor is made to be used with loaded runcards, in order to
+        minimize the amount of code needed to init a new object (you just to
+        load the runcard and call this function).
+
+        Note
+        ----
+        An object is initialized with no rotations, since the role of rotations
+        is to keep the current state of the output object after manipulations
+        happened.
+        Since a new object is here in the process of being created, no rotation
+        has to be logged.
+
+        Parameters
+        ----------
+        theory : dict
+            the theory card
+        operator : dict
+            the operator card
+        path : str or os.PathLike
+            the underlying path (if not provided, it is created in a temporary
+            path)
+
+        Returns
+        -------
+        EKO
+            the generated structure
+
+        """
+        path = pathlib.Path(
+            path if path is not None else tempfile.mkstemp(suffix=".tar")[1]
+        )
+        path.unlink()
```

Comment:
this needs to be confirmed by an external argument

Reply:
This one I solved in another way, but I guess it does what you wanted.
In any case, a more stable solution will come with #139